src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,98 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+import React, { useContext } from 'react';
+
+import { TodoContext } from '../../context/ContextTodo';
+
+import { ETodoStatus, Todo } from '../../types/Todo';
+
+interface ITodosQuery {
+  inputField?: string;
+  filteredBy?: ETodoStatus;
+}
+
+export function getPreparedTodos(todos: Todo[], {
+  inputField = '',
+  filteredBy = ETodoStatus.ALL,
+}: ITodosQuery) {
+  let preparedTodos = [...todos];
+
+  if (inputField) {
+    const searchQueryLowerCase = inputField.toLowerCase().trim();
+
+    preparedTodos = preparedTodos.filter(({ title }) => (
+      title.toLowerCase().includes(searchQueryLowerCase)
+    ));
+  }
+
+  if (filteredBy !== ETodoStatus.ALL) {
+    switch (filteredBy) {
+      case ETodoStatus.Active:
+        return preparedTodos.filter(({ completed }) => !completed);
+
+      case ETodoStatus.Completed:
+        return preparedTodos.filter(({ completed }) => completed);
+
+      default:
+        throw new Error('Invalid select status');
+    }
+  }
+
+  return preparedTodos;
+}
+
+export const TodoFilter = () => {
+  const {
+    inputField,
+    setInputField,
+    setFilteredBy,
+    filteredBy,
+  } = useContext(TodoContext);
+
+  const handleSelectStatus = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setFilteredBy(event.currentTarget.value as ETodoStatus);
+  };
+
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            data-cy="statusSelect"
+            value={filteredBy}
+            onChange={handleSelectStatus}
+          >
+            <option value="all">All</option>
+            <option value="active">Active</option>
+            <option value="completed">Completed</option>
```

Comment:
You have enum for this value, render options using map and keys from enum
https://stackoverflow.com/questions/41308123/map-typescript-enum

Reply:
Yes, that is a valid comment.